### PR TITLE
OEL-1873: Manage menu items role

### PR DESCRIPTION
**Pull request by escuriola**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: oe_showcase.post_update.php [hunk 1]
```diff
@@ -243,3 +243,22 @@ function oe_showcase_post_update_00010(): void {
...
+function oe_showcase_post_update_00011(): void {
+  $role = Role::create([
+    'id' => 'manage_menu_items',
+    'label' => 'Manage menu items',
+    'is_admin' => FALSE,
+    'permissions' => [
+      'access administration pages',
+      'access toolbar',
+      'administer menu',
+      'administer site configuration',
```
Comment (drishu) [inline oe_showcase.post_update.php]: dont think this is needed, gives to much access

File: tests/src/ExistingSite/AuthorisationTest.php [hunk 2]
```diff
@@ -91,6 +91,45 @@ public function testAuthenticatedAccess(): void {
...
+    $assert_session->pageTextContains('Menu Example menu has been added.');
+
+    $page->clickLink('Add link');
+    $page->fillField('Menu link title', 'Link 1');
+    $page->fillField('Link', 'https://europa.eu');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains('The menu link has been saved.');
+    $assert_session->pageTextContains('Link 1');
+  }
+
```
Comment (drishu) [inline tests/src/ExistingSite/AuthorisationTest.php:156]: in testManageUsersRoleAssign add this new role, same in testOtherRolesCannotManageUsers

File: tests/src/ExistingSite/AuthorisationTest.php [hunk 3]
```diff
@@ -91,6 +91,45 @@ public function testAuthenticatedAccess(): void {
...
+      '/admin/structure/menu/add',
+      '/admin/structure/menu/manage/main',
+      'admin/structure/menu/manage/main/add',
+    ];
+    foreach ($paths as $path) {
+      $this->drupalGet($path);
+      $this->assertSession()->statusCodeEquals(200);
+    }
+
+    // Mark test content for deletion after the test has finished.
```
Comment (drishu) [inline tests/src/ExistingSite/AuthorisationTest.php]: Starting from here it should go into a dedicated class file ManageMenuItemsTest.php

**Review by drishu — CHANGES_REQUESTED** [review]

File: oe_showcase.post_update.php [hunk 4]
```diff
@@ -243,3 +243,22 @@ function oe_showcase_post_update_00010(): void {
...
+      'view the administration theme',
+    ],
+  ]);
+  $role->save();
+}
```
Comment (escuriola) [inline oe_showcase.post_update.php:281]: Missing this:

  // Add new role to roleassign.
  $roleassign_config = \Drupal::configFactory()->getEditable('roleassign.settings');
  $roleassign_roles = $roleassign_config->get('roleassign_roles');
  $roleassign_roles['manage_menu_items'] = 'manage_menu_items';
  $roleassign_config->set('roleassign_roles', $roleassign_roles)->save();

File: tests/src/ExistingSite/ManageMenuItemsTest.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * Tests that a manage_menu_items user can manage menus.
+ */
+class ManageMenuItemsTest extends ShowcaseExistingSiteTestBase {
+
+  use UserTrait;
+
+  /**
+   * Test users with manage_menu_items role can manage menus.
+   */
+  public function testUsersWithManageSiteSpecificFooterRole(): void {
```
Comment (drishu) [inline tests/src/ExistingSite/ManageMenuItemsTest.php]:
```suggestion
  public function testUsersWithManageMenuItemsRole(): void {
```

**Review by drishu — CHANGES_REQUESTED** [review]
and pls resolve the conflicts

**Review by drishu — APPROVED** [review]

File: config/install/user.role.manage_menu_items.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - system
+    - toolbar
+id: manage_menu_items
+label: 'Manage menu items'
+weight: 6
+is_admin: null
```
Comment (donquixote) [inline config/install/user.role.manage_menu_items.yml:10]: Strange, how do we get `null` here instead of `false`?

Comment (donquixote) [reply config/install/user.role.manage_menu_items.yml:10]: Interesting, creating a role with drush sets this to `null`, but then updating through the UI sets it to `false`.
I think we can live with it.

**Review by donquixote — APPROVED** [review]
I noticed one thing but I think we can ignore it.
I could comment more on comments and names, but I rather see this merged.

**Pull request merged**